area.go

```diff
@@ -0,0 +1,258 @@
+}
+
+// ShowChild runs the show-child action
+func (c *AreaController) ShowChild(ctx *app.ShowChildAreaContext) error {
+	id, err := uuid.FromString(ctx.ID)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
+	}
+
+	return application.Transactional(c.db, func(appl application.Application) error {
+		parentArea, err := appl.Areas().Load(ctx, id)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		children, err := appl.Areas().ListChildren(ctx, parentArea)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+
+		res := &app.AreaList{}
+		res.Data = ConvertAreas(appl, ctx.RequestData, children, addResolvedPath)
+
+		return ctx.OK(res)
+	})
+}
+
+// CreateChild runs the create-child action.
+func (c *AreaController) CreateChild(ctx *app.CreateChildAreaContext) error {
+
+	_, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
+	}
+	parentID, err := uuid.FromString(ctx.ID)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
+	}
+
+	return application.Transactional(c.db, func(appl application.Application) error {
+		parent, err := appl.Areas().Load(ctx, parentID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
+		}
+
+		reqArea := ctx.Payload.Data
+		if reqArea.Attributes.Name == nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("data.attributes.name", nil).Expected("not nil"))
+		}
+
+		childPath := area.ConvertToLtreeFormat(parentID.String())
+		if parent.Path != "" {
+			childPath = parent.Path + pathSepInDatabase + childPath
+		}
+		newArea := area.Area{
+			SpaceID: parent.SpaceID,
+			Path:    childPath,
+			Name:    *reqArea.Attributes.Name,
+		}
+
+		err = appl.Areas().Create(ctx, &newArea)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+
+		res := &app.AreaSingle{
+			Data: ConvertArea(appl, ctx.RequestData, &newArea, addResolvedPath),
+		}
+		ctx.ResponseData.Header().Set("Location", rest.AbsoluteURL(ctx.RequestData, app.AreaHref(res.Data.ID)))
+		return ctx.Created(res)
+	})
+}
+
+// Show runs the show action.
+func (c *AreaController) Show(ctx *app.ShowAreaContext) error {
+	id, err := uuid.FromString(ctx.ID)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
+	}
+
+	return application.Transactional(c.db, func(appl application.Application) error {
+		a, err := appl.Areas().Load(ctx, id)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+
+		res := &app.AreaSingle{}
+		res.Data = ConvertArea(appl, ctx.RequestData, a, addResolvedPath)
+		return ctx.OK(res)
+	})
+}
+
+// addResolvedPath resolves the path in the form of /area1/area2/area3
+func addResolvedPath(appl application.Application, req *goa.RequestData, mArea *area.Area, sArea *app.Area) error {
```

Comment:
>  Do not talk to DB from Convert* functions. Perform all DB related transactions in repo level and call them from controllers.

Thanks @pranavgore09 , I did try to avoid that but realized that I would end up duplicating code across 6+ places.. and then also thought that ConvertArea should be handling the scenario of db <=> service format interchanges.